app/services/testdata/RegisteredStatusGenerator.scala

```diff
@@ -24,41 +24,38 @@ import services.testdata.faker.DataFaker._
 import uk.gov.hmrc.play.http.HeaderCarrier
 
 object RegisteredStatusGenerator extends RegisteredStatusGenerator {
-  override val appRepository = applicationRepository
-  override val pdRepository = personalDetailsRepository
-  override val qRepository = questionnaireRepository
+  override val authProviderClient = AuthProviderClient
 }
 
 trait RegisteredStatusGenerator extends BaseGenerator {
   import scala.concurrent.ExecutionContext.Implicits.global
 
-  val appRepository: GeneralApplicationRepository
-  val pdRepository: PersonalDetailsRepository
-  val qRepository: QuestionnaireRepository
+  val authProviderClient: AuthProviderClient.type
```

Comment:
Just a question: interesting use of Singleton.type here. What was the intention?